### ISSUE TITLE
child logger namespace collision ordering

### ISSUE THREAD

**Issue by davidmarkclements** [pinojs/pino#222]
I think this might (possibly) be a regression because I remember dealing with this in distant past

```sh
$ node -e "require('./')().child({a: 1}).child({a: 2}).info({a: 3})"
{"pid":8150,"hostname":"Davids-MacBook-Pro.local","level":30,"time":1490215591008,"a":3,"a":1,"a":2,"v":1}
```

We made a design decision early on that it's fine to repeat colliding property names in the JSON output, *as long as it was ordered correctly* - because once it gets `JSON.parse`d that last occurrence becomes the value in the object, e.g.:

```sh
 $ node -e "require('./')().child({a: 1}).child({a: 2}).info({a: 3})" | pino
[2017-03-22T20:46:33.238Z] INFO (8152 on Davids-MacBook-Pro.local):
    a: 2
```

The value of `a` should be 3, because log output should override child properties

**Comment by jsumners** [comment]
Fixed by #224

**Comment by rayvincent-bsd** [comment]
I just ran into this one which could get ugly... Take this as an example:

`logger.trace({level:50}, "This should be a trace, but instead logs as an error");`

This will log as an error.
